software/bsg_manycore_lib/bsg_manycore_arch.h

```diff
@@ -104,6 +104,16 @@
                         | ((((px+1)*bsg_global_X)+x) << GLOBAL_X_CORD_SHIFT)  \
                         | ((int) local_addr)))
 
+// physical pod remote pointer, corresponds to all pods for px, py
+// px = pod id x
+// py = pod id y
+// x = subcord x
+// y = subcord y
+#define bsg_global_physical_pod_ptr(px,py,x,y,local_addr) \
+  ((bsg_remote_int_ptr) ( (1 << GLOBAL_PREFIX_SHIFT)    \
+                        | ((((py)*bsg_global_Y)+y) << GLOBAL_Y_CORD_SHIFT)  \
+                        | (((px*bsg_global_X)+x) << GLOBAL_X_CORD_SHIFT)  \
```

Comment:
Maybe this hasn't become a problem yet, but these arguments need extra parentheses. if px was "1+1", that expression becomes (1+1*bsg_global_X), which is not the same as 2*bsg_global_X

Reply:
kk will fix! I'll go through the file and see if any other macros have similar mistakes